### PR TITLE
Grin 2020 roadmap

### PR DESCRIPTION
**Pull request by lehnberg**
[Rendered text](https://github.com/mimblewimble/grin-rfcs/blob/17ba9084f95fb93299ff1d63925c8f9b51e49dbe/text/0000-2020-roadmap.md)

Following discussions in governance meetings and the forum post asking for ideas[1], this RFC is a first attempt to produce a roadmap for 2020.

**PLEASE NOTE**
While the wording in the RFC might sound like it's already set in stone, this is a **first draft** based on my own _personal_ and _highly subjective_ opinions of what I think should be a priority in 2020.

In accordance with our RFC process, **review and discussion is very much welcomed** in this PR, and improvements will be made as required in an attempt to build consensus on some final list of contents.

[1] https://forum.grin.mw/t/grin2020-roadmap-calling-for-blog-posts-with-ideas/6327

### CODE REVIEW

File: text/0000-2020-roadmap.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+# Community-level explanation
+
+[community-level-explanation]: #community-level-explanation
+
+## Development
+
+- **Transaction building.** Establish an approach that will be robust and good enough be a viable baseline for building grin transactions, and can replace unsafe http methods.
+- **Output linkability.** Explore ideas and approaches to obfuscate the transaction graph, both in the mempool/p2p layer, and on the blockchain itself.
+- **P2P layer.** Iterate on dandelion, explore Erlay, explore encrypting node traffic, and determine whether I2P support is something that should be dropped or targeted for implementation.
+- **Syncing and IBD.** Explore FlyClient as a viable approach to light clients for mobiles and other devices, in order to reduce the reliance on hosted nodes.
```
Comment (tromp) [inline text/0000-2020-roadmap.md]: More important than FlyClient is:
Take advantage of unspent bitmap commitment added in HF to split IBD among peers to improve latency, speed, and reliability.

Comment (lehnberg) [reply text/0000-2020-roadmap.md]: Essentially getting rid of `txhashset.zip`? I completely agree.

Comment (tromp) [reply text/0000-2020-roadmap.md]: Either getting rid of it, or breaking it up into small pieces...

File: text/0000-2020-roadmap.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+- **Mining protocol.** Research Stratum v2 and establish whether it's a viable solution to reduce mining pool centralization.
```
Comment (tromp) [inline text/0000-2020-roadmap.md]: - **PoW.** Prepare final tweak of ASIC Resistant secondary PoW.

File: text/0000-2020-roadmap.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+Furthermore, the process can be helpful in and of itself. As participants attempt to agree on a list of items that they consider are important to target for next year, it also becomes clear what is _not as important_ in this context.
+
+# Community-level explanation
+
+[community-level-explanation]: #community-level-explanation
+
+## Development
+
+- **Transaction building.** Establish an approach that will be robust and good enough be a viable baseline for building grin transactions, and can replace unsafe http methods.
```
Comment (tromp) [inline text/0000-2020-roadmap.md]: - **Kernal types.** Complete specification of and implement relative kernels as outlined in https://github.com/mimblewimble/grin-rfcs/pull/19

**Comment by lehnberg** [comment]
@tromp thanks for review and feedback, I've added all your points in the doc. Will wait for others to comment and add thoughts and then perhaps it makes sense to try to shorten this list down a bit?

**Comment by johndavies24** [comment]
FYI, new address is https://forum.grin.mw/t/grin2020-roadmap-calling-for-blog-posts-with-ideas/6327

**Comment by yeastplume** [comment]
As of 11 Feb, 2020 this can be considered in Final Comment Period, and will be merged in two weeks. Please ensure any comments are made before then!

File: text/0000-2020-roadmap.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+*notguilty:*
+> Features are fun and nice but secondary to basic reliability. I would like to see a solid foundation...
+
+*Roelsmajor*
+> When [FlyClient paper] https://eprint.iacr.org/2019/226?
+
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
```
Comment (DavidBurkett) [inline text/0000-2020-roadmap.md]: We should probably make an effort to resolve some of these before merging this. Doesn't seem very useful without answering the 2nd one, for example.

**Comment by lehnberg** [comment]
Closing as per yesterday's meeting. Contents cross-published to https://forum.grin.mw/t/grin2020-roadmap/7096

**Pull request closed without merging**